IMS.BL/Application/Program.cs

```diff
@@ -62,19 +63,24 @@ public static void Main(string[] args)
                     case 4:
                     {
                         var productId = getProductData.GetProductId();
-                        var productString = inventoryRepository.SearchForOneProduct(productId);
+                        var product = inventoryRepository.GetOneProduct(productId);
                     
-                        Console.WriteLine($"The product you are searching for: {productString}");
+                        Console.WriteLine($"The product with id of: {product.Id} has a name of {product.Name}, its cost is {product.Price}, and {product.Quantity} products are available!");
                         break;
                     }
                     case 5:
                     {
                         var products = inventoryRepository.GetAllProducts();
+
+                        if (!products.Any())
+                        {
+                            throw new NullReferenceException();
```

Comment:
make a custom exception and use it on catch to deliver a custom message for the user